aws/resource_aws_cur_report_definition.go

```diff
@@ -0,0 +1,189 @@
+				ValidateFunc: validation.StringInSlice([]string{
+					costandusagereportservice.TimeUnitDaily,
+					costandusagereportservice.TimeUnitHourly,
+				}, false),
+			},
+			"format": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					costandusagereportservice.ReportFormatTextOrcsv}, false),
+			},
+			"compression": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					costandusagereportservice.CompressionFormatGzip,
+					costandusagereportservice.CompressionFormatZip,
+				}, false),
+			},
+			"additional_schema_elements": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						costandusagereportservice.SchemaElementResources,
+					}, false),
+				},
+				Set:      schema.HashString,
+				Required: true,
+				ForceNew: true,
+			},
+			"s3_bucket": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"s3_prefix": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(0, 256),
+			},
+			"s3_region": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"additional_artifacts": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						costandusagereportservice.AdditionalArtifactQuicksight,
+						costandusagereportservice.AdditionalArtifactRedshift,
+					}, false),
+				},
+				Set:      schema.HashString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAwsCurReportDefinitionCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).costandusagereportconn
+
+	reportName := *aws.String(d.Get("report_name").(string))
+
+	reportDefinition := &costandusagereportservice.ReportDefinition{
+		ReportName:  &reportName,
+		TimeUnit:    aws.String(d.Get("time_unit").(string)),
+		Format:      aws.String(d.Get("format").(string)),
+		Compression: aws.String(d.Get("compression").(string)),
+		S3Bucket:    aws.String(d.Get("s3_bucket").(string)),
+		S3Prefix:    aws.String(d.Get("s3_prefix").(string)),
+		S3Region:    aws.String(d.Get("s3_region").(string)),
+	}
+
+	additionalSchemaElementsSet := d.Get("additional_schema_elements").(*schema.Set)
+	for _, additionalSchemaElement := range additionalSchemaElementsSet.List() {
+		reportDefinition.AdditionalSchemaElements = append(reportDefinition.AdditionalSchemaElements,
+			aws.String(additionalSchemaElement.(string)))
+	}
+
+	additionalArtifactsSet := d.Get("additional_artifacts").(*schema.Set)
+	for _, additionalArtifact := range additionalArtifactsSet.List() {
+		reportDefinition.AdditionalArtifacts = append(reportDefinition.AdditionalArtifacts,
```

Comment:
It may be worth seeing if all this `additionalArtifactsSet` logic and the loop can be removed with:

```go
AdditionalArtifacts: expandStringSet(d.Get("additional_artifacts").(*schema.Set)),
```

in the `reportDefinition` above.